icelake/src/io/file_writer/position_delete_writer.rs

```diff
@@ -0,0 +1,198 @@
+//! A module provide `PositionDeleteWriter`.
+use std::collections::BTreeMap;
+use std::sync::Arc;
+
+use crate::config::TableConfigRef;
+use crate::io::location_generator::FileLocationGenerator;
+use crate::types::{Any, DataFileBuilder, Field, Primitive, Schema};
+use crate::{types::Struct, Result};
+use crate::{Error, ErrorKind};
+use arrow_array::{ArrayRef, Int64Array, RecordBatch, StringArray};
+use arrow_schema::SchemaRef;
+use opendal::Operator;
+
+use super::rolling_writer::RollingWriter;
+
+/// A PositionDeleteWriter used to write position delete, it will sort the incoming delete by file_path and pos.
+pub struct SortedPositionDeleteWriter {
+    operator: Operator,
+    table_location: String,
+    location_generator: Arc<FileLocationGenerator>,
+    table_config: TableConfigRef,
+
+    delete_cache: BTreeMap<String, Vec<i64>>,
```

Comment:
Java version uses roaring bitmap, which saves a lot of memory, how about we also use it?

Reply:
Roaring bitmap looks good to me. How about put it in a later refine PR.🤔

Reply:
ok